<html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="This guide shows you how to use the IDS Wizard.
"><link href=https://uoa-eresearch.github.io/cer-documentation/Research%20Data/Data%20Storage/IDS/Wizard/Tutorial/guide/ rel=canonical><link rel=icon href=../../../../../../assets/favicon.ico><meta name=generator content="mkdocs-1.5.3, mkdocs-material-8.1.3"><title>Using the IDS Wizard - CeR Documentation</title><link rel=stylesheet href=../../../../../../assets/stylesheets/main.fad675c6.min.css><link rel=stylesheet href=../../../../../../assets/stylesheets/palette.356b1318.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7C&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:""}</style><link rel=stylesheet href=../../../../../../assets/_mkdocstrings.css><link rel=stylesheet href=../../../../../../stylesheets/extras.css><script>__md_scope=new URL("../../../../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><style>
main{
	transition:opacity ease-in 0.2s
}
 [hidden]{
	display:none !important;
}
[unresolved] {
    opacity: 0;
    display: block;
    overflow: hidden;
    position: relative;
    margin: 0;
}
[dir="ltr"] .md-typeset :is(.admonition-title, summary) {
  border-left-width: 0;
}
body[dir="ltr"] .ltr-flip {
  transform:scaleX(-1)
}
body[dir="rtl"] .ltr-flip {
  transform: none
}
header form{
  margin-bottom:0
}

    :root{
  --sidebar-width:14rem;
  --sidebar-offset:-14.1rem;
}
header.md-header, [class^="md-tabs"]{
  max-height:2.4rem;
  overflow-y:hidden;
}
.main-height{
  height:calc(100vh - 2.4rem)
}
.main-height.tabs{
  height:calc(100vh - 4.8rem)
}
.md-header__topic:first-child {
  font-weight: normal;
}
.md-nav{
  font-size: 0.9rem;
}
.md-sidebar{
  align-self:flex-start;
  flex-shrink:0;
  padding:1.2rem 0;
  position:sticky;
  top:2.4rem;
  width:var(--sidebar-width);
}
.md-source-file{
  display:flex;
  justify-content:space-between;
  align-items:center;
}

@media screen and (max-width:76.1875em){
   [dir=ltr] .md-sidebar--primary{
      left:var(--sidebar-offset);
  }
  [dir=rtl] .md-sidebar--primary{
      right:var(--sidebar-offset);
  }
   .md-sidebar--primary{
      background-color:var(--md-default-bg-color);
      display:block;
      height:100%;
      position:fixed;
      top:0;
      transform:translateX(0);
      transition:transform .25s cubic-bezier(.4,0,.2,1),box-shadow .25s;
      width:var(--sidebar-width);
      z-index:5
  }
   [data-md-toggle=drawer]:checked~.md-container .md-sidebar--primary{
      box-shadow:var(--md-shadow-z3);
      transform:translateX(var(--sidebar-width))
  }
   [dir=rtl] [data-md-toggle=drawer]:checked~.md-container .md-sidebar--primary{
      transform:translateX(var(--sidebar-offset)) 
  }
  .main-height.tabs{
    height:calc(100vh - 2.4rem)
  }
}


@media print{
  header.md-hader{
		display:none;
	}
  section:not(:first-child) {
    page-break-before:always;
  }
  .main-height, .main-height.tabs{
    height: initial;
  }
  .headerlink{
    display:none;
  }
  .md-container{
    display:block;
  }
  .md-main__inner{display:block}
  .md-sidebar{display:none;}
  .md-source-file{
    display: inline-block;
  }
}
  </style><link rel=stylesheet href=../../../../../../assets/stylesheets/codelab_bin.css></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary=custom data-md-color-accent=deep-purple> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#annotating-your-first-dataset class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class="md-header md-header--lifted" data-md-component=header> <nav class=md-header__inner aria-label=Header> <a href class="md-header__button md-logo" title="Exit to Home"> <div class=ltr-flip><svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 3H5c-1.11 0-2 .89-2 2v4h2V5h14v14H5v-4H3v4a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2m-8.92 12.58L11.5 17l5-5-5-5-1.42 1.41L12.67 11H3v2h9.67l-2.59 2.58Z"></path></svg></div> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M7 13v-2h14v2H7m0 6v-2h14v2H7M7 7V5h14v2H7M3 8V5H2V4h2v4H3m-1 9v-1h3v4H2v-1h2v-.5H3v-1h1V17H2m2.25-7a.75.75 0 0 1 .75.75c0 .2-.08.39-.21.52L3.12 13H5v1H2v-.92L4 11H2v-1h2.25Z"></path></svg> </label> <div class=md-header__title> <div class=md-header__topic> <span class=md-ellipsis> Using the IDS Wizard </span> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media data-md-color-scheme=default data-md-color-primary=custom data-md-color-accent=deep-purple aria-label="Switch to dark mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_2 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M7.5 2c-1.79 1.15-3 3.18-3 5.5s1.21 4.35 3.03 5.5C4.46 13 2 10.54 2 7.5A5.5 5.5 0 0 1 7.5 2m11.57 1.5 1.43 1.43L4.93 20.5 3.5 19.07 19.07 3.5m-6.18 2.43L11.41 5 9.97 6l.42-1.7L9 3.24l1.75-.12.58-1.65L12 3.1l1.73.03-1.35 1.13.51 1.67m-3.3 3.61-1.16-.73-1.12.78.34-1.32-1.09-.83 1.36-.09.45-1.29.51 1.27 1.36.03-1.05.87.4 1.31M19 13.5a5.5 5.5 0 0 1-5.5 5.5c-1.22 0-2.35-.4-3.26-1.07l7.69-7.69c.67.91 1.07 2.04 1.07 3.26m-4.4 6.58 2.77-1.15-.24 3.35-2.53-2.2m4.33-2.7 1.15-2.77 2.2 2.54-3.35.23m1.15-4.96-1.14-2.78 3.34.24-2.2 2.54M9.63 18.93l2.77 1.15-2.53 2.19-.24-3.34Z"></path></svg> </label> <input class=md-option data-md-color-media data-md-color-scheme=slate data-md-color-primary=custom data-md-color-accent=light-green aria-label="Switch to light mode" type=radio name=__palette id=__palette_2> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_1 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M7.5 2c-1.79 1.15-3 3.18-3 5.5s1.21 4.35 3.03 5.5C4.46 13 2 10.54 2 7.5A5.5 5.5 0 0 1 7.5 2m11.57 1.5 1.43 1.43L4.93 20.5 3.5 19.07 19.07 3.5m-6.18 2.43L11.41 5 9.97 6l.42-1.7L9 3.24l1.75-.12.58-1.65L12 3.1l1.73.03-1.35 1.13.51 1.67m-3.3 3.61-1.16-.73-1.12.78.34-1.32-1.09-.83 1.36-.09.45-1.29.51 1.27 1.36.03-1.05.87.4 1.31M19 13.5a5.5 5.5 0 0 1-5.5 5.5c-1.22 0-2.35-.4-3.26-1.07l7.69-7.69c.67.91 1.07 2.04 1.07 3.26m-4.4 6.58 2.77-1.15-.24 3.35-2.53-2.2m4.33-2.7 1.15-2.77 2.2 2.54-3.35.23m1.15-4.96-1.14-2.78 3.34.24-2.2 2.54M9.63 18.93l2.77 1.15-2.53 2.19-.24-3.34Z"></path></svg> </label> </form> <div class=md-header__source> <a href=https://github.com/UoA-eResearch/cer-documentation title=source class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><!-- Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"></path></svg> </div> <div class=md-source__repository> cer-documentation </div> </a> </div> </nav> <nav class=md-tabs aria-label=tabs data-md-component=tabs> <div class=md-grid> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=../../../../../.. class=md-tabs__link> Start here </a> </li> <li class=md-tabs__item> <a href=../../../../../../Research%20Compute/01ResearchCompute/ class=md-tabs__link> Research Compute </a> </li> <li class="md-tabs__item md-tabs__item--active"> <a href=../../../../../ResearchDataOverview/ class=md-tabs__link> Research Data </a> </li> <li class=md-tabs__item> <a href=../../../../../../Research%20Visualisation/ResearchVis/ class=md-tabs__link> Research Visualisation </a> </li> <li class=md-tabs__item> <a href=../../../../../../codelab/ class=md-tabs__link> Codelab </a> </li> <li class=md-tabs__item> <a href=../../../../../../projects/DocumentationArchitecture/ class=md-tabs__link> Projects </a> </li> </ul> </div> </nav> </header> <div class=md-container data-md-component=container> <main class=md-main data-md-component=main unresolved> <div class="md-main__inner " style=margin-top:0;> <div class="md-sidebar md-sidebar--primary" data-md-type=toc style=padding:0;> <div id=drawer class="md-sidebar__scrollwrap md-sidebar__inner main-height 
  
"> <nav class="steps md-nav md-nav--secondary" aria-label="Table of contents"> <ol class=md-nav__list data-md-component=toc data-md-scrollfix> <li completed> <a href=#annotating-your-first-dataset> <span class=step> <span>Annotating your first dataset¶</span> </span> </a> </li> <li> <a href=#decide-how-to-describe-and-structure-your-data> <span class=step> <span>Decide how to describe and structure your data</span> </span> </a> </li> <li> <a href=#import-your-data-into-the-instrument-data-wizard> <span class=step> <span>Import your data into the Instrument Data Wizard</span> </span> </a> </li> <li> <a href=#annotate-your-data-with-metadata> <span class=step> <span>Annotate your data with metadata</span> </span> </a> </li> <li> <a href=#restrict-access-to-your-data> <span class=step> <span>Restrict access to your data</span> </span> </a> </li> <li> <a href=#save-the-ingestion-file> <span class=step> <span>Save the ingestion file</span> </span> </a> </li> </ol> </nav> <div class=metadata> <a target=_blank href><span class=md-icon> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M14 12h-4v-2h4m0 6h-4v-2h4m6-6h-2.81a5.985 5.985 0 0 0-1.82-1.96L17 4.41 15.59 3l-2.17 2.17a6.002 6.002 0 0 0-2.83 0L8.41 3 7 4.41l1.62 1.63C7.88 6.55 7.26 7.22 6.81 8H4v2h2.09c-.05.33-.09.66-.09 1v1H4v2h2v1c0 .34.04.67.09 1H4v2h2.81c1.04 1.79 2.97 3 5.19 3s4.15-1.21 5.19-3H20v-2h-2.09c.05-.33.09-.66.09-1v-1h2v-2h-2v-1c0-.34-.04-.67-.09-1H20V8Z"></path></svg> </span>Report a mistake</a> </div> </div> </div> <div id=steps class="md-typeset main-height 
    
  " label=Using-the-IDS-Wizard> <article class=md-content data-md-component=content> <section class=labstep data-step=1 selected><div id=codelab-about> <h1>Using the IDS Wizard</h1> <p>This guide shows you how to use the IDS Wizard.</p> <hr> <aside style="display:flex;justify-content: space-between;flex-wrap: wrap;"> <div style="display: flex;flex-flow: row nowrap;align-items: flex-start;color:var(--md-primary-bg-color--light);"> <span class=md-icon style="margin:0 0.5rem;"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 20a8 8 0 0 0 8-8 8 8 0 0 0-8-8 8 8 0 0 0-8 8 8 8 0 0 0 8 8m0-18a10 10 0 0 1 10 10 10 10 0 0 1-10 10C6.47 22 2 17.5 2 12A10 10 0 0 1 12 2m.5 5v5.25l4.5 2.67-.75 1.23L11 13V7h1.5Z"></path></svg></span>Estimated completion time: <span id=time-estimate></span> </div> <div class=md-source-file style="display:flex;flex-flow:column nowrap;align-items: end;"> <a href=https://github.com/UoA-eResearch/cer-documentation/edit/main/docs/Research%20Data/Data%20Storage/IDS/Wizard/Tutorial/guide.md title="Edit this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"></path></svg> </a> </div> </aside> </div> <div class=instructions><h2 duration=2:00 id=annotating-your-first-dataset class=step-title>1. Annotating your first dataset¶<a class=headerlink href=#annotating-your-first-dataset title="Permanent link">¶</a></h2> <p>Welcome! This tutorial will help you annotate a dataset using the Instrument Data Wizard, in preparation for ingestion into Instrument Data Service.</p> <div class="admonition example"> <p class=admonition-title>Example</p> <p>This tutorial will follow a fictional scenario. You are Sarah <span class=twemoji><svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M6 17c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1H6m9-9a3 3 0 0 1-3 3 3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3M3 5v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2H5a2 2 0 0 0-2 2Z"></path></svg></span>, a PhD student collecting data for your cancer genomics research, and you would like to store that data in your University's instrument data repository.</p> </div> <div class="admonition info"> <p class=admonition-title>What is the Instrument Data Wizard?</p> <p>If you're not familiar with what Instrument Data Service and Instrument Data Wizard do, and how they work together, check out <a href=../../03-data-dictionary/ >this</a> first.</p> </div> <p>By the end of this tutorial, you will learn about the Instrument Data Wizard and Instrument Data Service, and how to:</p> <ul> <li>decide on a structure for your data for Instrument Data Service.</li> <li>import data files into the Instrument Data Wizard and add structure to it.</li> <li>annotate your data with metadata</li> <li>restrict access to your data or subset of the data</li> <li>export the metadata</li> </ul> <p><strong>If you want to follow along on your computer</strong>, please install Instrument Data Wizard, download the tutorial data, and unzip it on your desktop. This data will be used in the rest of the tutorial.</p> <p><a href=../../../assets/tutorial-data.zip><span class=twemoji><svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 6a2 2 0 0 1 2 2v10a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V6c0-1.11.89-2 2-2h6l2 2h8m-.75 7H16V9h-2v4h-3.25L15 17.25"></path></svg></span> Download the tutorial-data.zip file here</a></p> <p>Contact <a href=mailto:c.seal@auckland.ac.nz>Chris Seal</a> if you need assistance getting set up on Instrument Data Service.</p> <p>Ready? Let's begin!</p> </div></section><section class=labstep data-step=2><div class=instructions><h2 id=decide-how-to-describe-and-structure-your-data class=step-title>2. Decide how to describe and structure your data<a class=headerlink href=#decide-how-to-describe-and-structure-your-data title="Permanent link">¶</a></h2> <p>Before we start using the Instrument Data Wizard, it's good to plan out and document <strong>how your data will be structured</strong> for Instrument Data Service, so you and your collaborators (and your future self!) will easily be able to find and use your data in the Instrument Data Service web portal.</p> <h3 id=how-instrument-data-service-structures-data>How Instrument Data Service structures data<a class=headerlink href=#how-instrument-data-service-structures-data title="Permanent link">¶</a></h3> <p>At its heart, Instrument Data Service is a database. Data are organised in a hierarchical structure, like folders and subfolders. There are three levels. You need to decide how to structure your data to fit in with this hierarchy.</p> <p><img alt=image src=../../../assets/OnionModel.svg width=300px></p> <ul> <li>Data files are grouped into <em>Datasets</em>.</li> <li><em>Datasets</em> are organised into <em>Experiments</em>.</li> <li><em>Experiments</em> belong to a <em>Project</em>.</li> </ul> <p>A <em>Dataset</em> may belong to multiple <em>Experiments</em>, and an <em>Experiment</em> may belong to multiple <em>Projects</em>.</p> <p>At each level of the hierarchy and at the individual file level, there are mandatory metadata fields that you can use to describe your data. There is also the ability to associate a custom metadata schema at each level, which allows you to record any relevant domain-specific observations and variables. The Instrument Data Service Search functionality allows you to filter for data based on metadata.</p> <h3 id=things-to-consider>Things to consider<a class=headerlink href=#things-to-consider title="Permanent link">¶</a></h3> <p>Here are some things to consider when deciding how your data should fit into this hierarchy:</p> <p><img alt=image height=300px src=../../../assets/OnionModel2.svg></p> <ul> <li>Usually, there should be one <em>Project</em> that corresponds to the research project or unit of research activity that you are collecting data for.</li> <li>Use <em>Experiments</em> to represent a study sample or variable you are studying. Store properties about the sample as metadata in each <em>Experiment</em>.</li> <li>Create one <em>Dataset</em> for each instrument you are using for acquiring data. Use the <em>Dataset</em> to contain all data files from that instrument for the <em>Experiment</em>. Store instrument run conditions as metadata in each <em>Dataset</em>.</li> <li>If the data is already using a directory structure, consider how that could translate into the hierarchical groupings.</li> <li>Consider whether you need to restrict access for a subset of your data. If so, you can group them as separate <em>Experiments</em> or <em>Datasets</em>. Later, you can restrict access to them as a whole group.</li> </ul> <h3 id=sarahs-sequencing-data>Sarah's sequencing data<a class=headerlink href=#sarahs-sequencing-data title="Permanent link">¶</a></h3> <p>As Sarah, you are studying how breast cancer cells respond to different drug treatments. You work closely within a group of three other PhD students led by Dr Charlotte Henare.</p> <p>To collect data, you take samples of cancer cells from an anonymous donor from the Hospital, treating the samples with drugs, and sending them off to a sequencing company for processing.</p> <p>You sent off 15 cell samples (5 for each group) six weeks ago that have:</p> <ol> <li>no treatment,</li> <li>treatment with the drug Herceptin, and</li> <li>treatment with the drug Keytruda.</li> </ol> <p>The tutorial data you downloaded are the resulting files sent through a USB stick via courier. The data contains the raw <code>*.fastq</code> files as well as aligned <code>*.bam</code> files (data processed by a bioinformatician) for each kind of treatment. </p> <div class="admonition example"> <p class=admonition-title>Example</p> <p>You are now responsible for organising, setting up access control and adding metadata to the data, before it is ingested into the University's online instrument data repository Instrument Data Service . You and your group members can then access the data through the repository portal and use it in analysis.</p> </div> <p>Because the raw data is sensitive and could lead to the identification of patients, any raw data should only be accessible to yourself. Other data can be accessed by the whole group (with the group ID <code>eres004011-admin</code>)</p> <p>After a discussion with your collaborators, you have created this data structure plan.</p> <h4 id=sample-data-structure-plan>Sample data structure plan<a class=headerlink href=#sample-data-structure-plan title="Permanent link">¶</a></h4> <ul> <li><strong>Project</strong> - The Project is called “Breast Cancer Drug Treatment Genomics” project, with the ID “BREAST04”.</li> <li><strong>Experiments</strong> - One experiment for each treatment type (i.e. “No treatment” with ID “NoTreatment”, “Herceptin” with ID “Herceptin”, Keytruda with ID “Keytruda”).</li> <li><strong>Datasets</strong> - In each Experiment, there would be two <em>Datasets</em>: one for raw <code>.fastq</code> files named “Raw” with ID “[Treatment ID]-Raw”, and another for aligned <code>.bam</code> files named “Aligned” with ID “[Treatment ID]-Aligned”.</li> <li><strong>Datafiles</strong> - Under each Dataset, there would be five files, one file from each tissue sample.</li> <li>Clinical details and sequencing instrument configurations will be recorded as metadata at the Dataset level.</li> </ul> <div class=highlight><pre><span></span><code><a id=__codelineno-0-1 name=__codelineno-0-1 href=#__codelineno-0-1></a>Project: “Breast Cancer Drug Treatment Genomics” - ID “BREAST04”/
<a id=__codelineno-0-2 name=__codelineno-0-2 href=#__codelineno-0-2></a>├─ Experiments: (One experiment per treatment)/
<a id=__codelineno-0-3 name=__codelineno-0-3 href=#__codelineno-0-3></a>│  ├─ “No treatment” with ID “NoTreatment”/
<a id=__codelineno-0-4 name=__codelineno-0-4 href=#__codelineno-0-4></a>│  │  ├─ Datasets/
<a id=__codelineno-0-5 name=__codelineno-0-5 href=#__codelineno-0-5></a>│  │  │  ├─ RAW (for raw .fastq files) with ID “[Treatment ID]-Raw"/
<a id=__codelineno-0-6 name=__codelineno-0-6 href=#__codelineno-0-6></a>│  │  │  │  ├─ File 1
<a id=__codelineno-0-7 name=__codelineno-0-7 href=#__codelineno-0-7></a>│  │  │  │  ├─ File 2
<a id=__codelineno-0-8 name=__codelineno-0-8 href=#__codelineno-0-8></a>│  │  │  │  ├─ File 3
<a id=__codelineno-0-9 name=__codelineno-0-9 href=#__codelineno-0-9></a>│  │  │  │  ├─ File 4
<a id=__codelineno-0-10 name=__codelineno-0-10 href=#__codelineno-0-10></a>│  │  │  │  ├─ File 5
<a id=__codelineno-0-11 name=__codelineno-0-11 href=#__codelineno-0-11></a>│  │  │  ├─ Aligned for aligned .bam files with ID “[Treatment ID]-Aligned”
<a id=__codelineno-0-12 name=__codelineno-0-12 href=#__codelineno-0-12></a>│  │  │  │  ├─ File 1
<a id=__codelineno-0-13 name=__codelineno-0-13 href=#__codelineno-0-13></a>│  │  │  │  ├─ ...
<a id=__codelineno-0-14 name=__codelineno-0-14 href=#__codelineno-0-14></a>│  ├─ "Keytruda" (with ID “Keytruda”)/
<a id=__codelineno-0-15 name=__codelineno-0-15 href=#__codelineno-0-15></a>│  │  ├─ Datasets/
<a id=__codelineno-0-16 name=__codelineno-0-16 href=#__codelineno-0-16></a>│  │  │  ├─ RAW (for raw .fastq files) with ID “[Treatment ID]-Raw"/
<a id=__codelineno-0-17 name=__codelineno-0-17 href=#__codelineno-0-17></a>│  │  │  │  ├─ File 1
<a id=__codelineno-0-18 name=__codelineno-0-18 href=#__codelineno-0-18></a>│  │  │  │  ├─ ...
<a id=__codelineno-0-19 name=__codelineno-0-19 href=#__codelineno-0-19></a>│  │  │  ├─ Aligned for aligned .bam files with ID “[Treatment ID]-Aligned”
<a id=__codelineno-0-20 name=__codelineno-0-20 href=#__codelineno-0-20></a>│  │  │  │  ├─ File 1
<a id=__codelineno-0-21 name=__codelineno-0-21 href=#__codelineno-0-21></a>│  │  │  │  ├─ ...
<a id=__codelineno-0-22 name=__codelineno-0-22 href=#__codelineno-0-22></a>│  ├─ “Herceptin” with ID “Herceptin”/
<a id=__codelineno-0-23 name=__codelineno-0-23 href=#__codelineno-0-23></a>│  │  ├─ Datasets/
<a id=__codelineno-0-24 name=__codelineno-0-24 href=#__codelineno-0-24></a>│  │  │  ├─ RAW (for raw .fastq files) with ID “[Treatment ID]-Raw"/
<a id=__codelineno-0-25 name=__codelineno-0-25 href=#__codelineno-0-25></a>│  │  │  │  ├─ File 1
<a id=__codelineno-0-26 name=__codelineno-0-26 href=#__codelineno-0-26></a>│  │  │  │  ├─ ...
<a id=__codelineno-0-27 name=__codelineno-0-27 href=#__codelineno-0-27></a>│  │  │  ├─ Aligned for aligned .bam files with ID “[Treatment ID]-Aligned”
<a id=__codelineno-0-28 name=__codelineno-0-28 href=#__codelineno-0-28></a>│  │  │  │  ├─ File 1
<a id=__codelineno-0-29 name=__codelineno-0-29 href=#__codelineno-0-29></a>│  │  │  │  ├─ ...
<a id=__codelineno-0-30 name=__codelineno-0-30 href=#__codelineno-0-30></a>
<a id=__codelineno-0-31 name=__codelineno-0-31 href=#__codelineno-0-31></a>(Clinical details and sequencing instrument configuration)
</code></pre></div> <div class="admonition example"> <p class=admonition-title>Exercise: How does your own data fit into this hierarchy?</p> <p>Think about the data you would like to ingest into Instrument Data Service and discuss it with your collaborators. Plan out how you would structure the data. Ask for a consultation with the friendly Instrument Data Service staff if you would like some help!</p> </div> </div></section><section class=labstep data-step=3><div class=instructions><h2 id=import-your-data-into-the-instrument-data-wizard class=step-title>3. Import your data into the Instrument Data Wizard<a class=headerlink href=#import-your-data-into-the-instrument-data-wizard title="Permanent link">¶</a></h2> <p>To import your data into the Instrument Data Wizard, open the Wizard and click the <code>Import data files</code> button.</p> <p><img alt=image src=../../../assets/import-1.png></p> <p>A step-by-step wizard will show up, with the first page giving a brief introduction to the Instrument Data Service hierarchy.</p> <p><img alt=image src=../../../assets/import-2.png></p> <h3 id=add-the-first-project-experiment-dataset-and-datafiles>Add the first Project, Experiment, Dataset and Datafiles<a class=headerlink href=#add-the-first-project-experiment-dataset-and-datafiles title="Permanent link">¶</a></h3> <p>Click <em>Next</em>. In the subsequent dialogs, the wizard will ask you where you would like to organise your data. Because we are starting with a blank metadata file, the wizard will ask you to create a new Project, Experiment and Dataset. </p> <p>As Sarah, you would like to import your raw data for the Keytruda trial. See if you can re-create this hierarchy in the Instrument Data Wizard:</p> <table> <thead> <tr> <th>Level</th> <th>Cell</th> <th>Input text</th> </tr> </thead> <tbody> <tr> <td>Project</td> <td>Proiect name</td> <td>Breast Cancer Drug Treatment Genomics</td> </tr> <tr> <td></td> <td>Proiect identifier</td> <td>BREAST04</td> </tr> <tr> <td></td> <td>Principal Investigator</td> <td>UPI123</td> </tr> <tr> <td></td> <td>Description</td> <td>My nice description of the project.</td> </tr> <tr> <td>Experiment</td> <td>Experiment name</td> <td>Keytruda</td> </tr> <tr> <td></td> <td>Experiment identifier</td> <td>Keytruda</td> </tr> <tr> <td></td> <td>Description</td> <td>My nice description of the experiment.</td> </tr> <tr> <td>Dataset</td> <td>Dataset name</td> <td>Raw</td> </tr> <tr> <td></td> <td>Dataset identifier</td> <td>Keytruda-Raw</td> </tr> <tr> <td></td> <td>Instrument identifier</td> <td>BiruVSlide1</td> </tr> <tr> <td>Datafiles</td> <td>all .fastq files <code>tutorial-data/keytruda/</code>.</td> <td></td> </tr> </tbody> </table> <p>The instrument ID (the persistent identifier, <code>PIDINST</code>) will be provided by the Facility Manager.</p> <p>Once finished, your editor should look like this.</p> <p><img alt=image src=../../../assets/import-3.png></p> <div class="admonition note"> <p class=admonition-title>What if I want to add data into an existing <code>Project</code> in the repository?</p> <p>If there is already a Project for your data in the data repository, you need to import it into the Instrument Data Wizard before you can start adding files to it. You can follow the instructions for <a href=../../02-adding-to-existing-structure/ >Adding data to existing Projects, Experiments or Datasets</a>.</p> </div> <h3 id=add-more-data>Add more data<a class=headerlink href=#add-more-data title="Permanent link">¶</a></h3> <p>What if you have files you need to organise separately from the initial import? Or if you need to add more files into the same dataset? For example, you may wish to ingest more than one sample or instrument run data files.</p> <p>You can click the <code>Import data files</code> button again, and the step-by-step wizard will prompt you to add files and ask how you would like to organise them.</p> <p><img alt=image src=../../../assets/AddDataAgain.jpg></p> <p>You can also right-click on the <code>Project</code>, <code>Experiment</code> or <code>Dataset</code> you would like to add more data to, and select the <code>Add Experiment</code>, <code>Add Dataset</code> or <code>Add files</code> options.</p> <p>As Sarah, you also have some raw data in the Herceptin trial you would like to import. </p> <p>After clicking the <code>Import data files</code> button and going through the initial explanation screen, you will now be presented with a choice to add files to an existing <code>Project</code>, or create a new <code>Project</code>. </p> <p><img alt=image src=../../../assets/import-4.png></p> <p>Since this is data for the same project, choose the "Breast Cancer Drug Treatment Genomics" <code>Project</code>.</p> <p>Then proceed through the rest of the wizard using this setup.</p> <ul> <li>Create a new Experiment with name "Herceptin", and ID "Herceptin".</li> <li>Create a new Dataset with the name "Raw", and ID "Herceptin-Raw".</li> <li>Data files: Add the <code>.fastq</code> files in the tutorial data folder, under <code>tutorial/herceptin/</code>. Once finished, your editor should look like this.</li> </ul> <p><img alt=image src=../../../assets/import-5.png></p> <h3 id=save-your-progress>Save your progress<a class=headerlink href=#save-your-progress title="Permanent link">¶</a></h3> <p>Instrument Data Wizard keeps all your data structure and annotations in an YAML-formatted ingestion file. This file is read by the Instrument Data Service ingestion process to find all your data files. It needs to be saved in the root folder of your data.</p> <p>Click the <span class=keys><kbd>Save</kbd></span> button, and save your ingestion file under the tutorial data folder. Name it <code>ingestion.yaml</code>. </p> <div class="admonition note"> <p class=admonition-title>Note</p> <p>Remember to save your changes as you work! As the Instrument Data Wizard is still being developed, bugs and crashes may happen at inopportune moments. After a crash, you can reopen the file using the <code>Open</code> button.</p> </div> <h3 id=exercise-add-even-more-data>Exercise: Add even more data<a class=headerlink href=#exercise-add-even-more-data title="Permanent link">¶</a></h3> <p>Try to re-create the hierarchy in the Instrument Data Wizard as described in the :ref:<code>example data structure plan &lt;sample-data-structure-plan&gt;</code>.</p> <p>Once finished, your editor should look like this.</p> <p><img alt=image src=../../../assets/import-exercise.png></p> </div></section><section class=labstep data-step=4><div class=instructions><h2 id=annotate-your-data-with-metadata class=step-title>4. Annotate your data with metadata<a class=headerlink href=#annotate-your-data-with-metadata title="Permanent link">¶</a></h2> <p>Now that your files are imported and organised, you can start annotating them. This step is vital to make the most use of the Instrument Data Service.</p> <h3 id=basic-metadata-fields>Basic metadata fields<a class=headerlink href=#basic-metadata-fields title="Permanent link">¶</a></h3> <p>In Instrument Data Service, there is a basic set of metadata fields applicable for any Projects, Experiments, Datasets or Datafiles. They are fields like <code>name</code>, <code>ID</code>, <code>author</code> and <code>institution</code>.</p> <h3 id=additional-metadata-with-schemas>Additional metadata with Schemas<a class=headerlink href=#additional-metadata-with-schemas title="Permanent link">¶</a></h3> <p>In addition, you can attach more metadata to a Project, Experiment, Dataset or Datafile through <code>Schemas</code>. They are made up of custom metadata fields called <code>Parameters</code>. You can specify the <code>Parameter</code> name and the value data type (for example, you can restrict the value to be a number, a string of characters, or a date.) </p> <p><code>Schemas</code> need to be defined in Instrument Data Service before you can use them in Instrument Data Wizard. </p> <p>A Project, Experiment, Dataset or Datafile can have <strong>multiple</strong> <code>Schemas</code> associated with them.</p> <div class="admonition question"> <p class=admonition-title>What can I store in Schemas?</p> <p>You can associate domain- or instrument-specific metadata with a Project, Experiment, Dataset or Datafile using <code>Schemas</code>. One way to use <code>Schema</code> could be to describe the study or treatment you have applied to the sample. Alternatively, you may wish to note down the instrument configuration used for acquiring data. For example, data from a sequencer may benefit from a <code>Schema</code> with depth of sequencing and sequencing method as <code>Parameters</code>.</p> <p>It's best to create a data dictionary document with your collaborators to specify what metadata should be stored. See <a href=../../03-data-dictionary/ >data dictionary</a>. </p> <p>As part of onboarding, Instrument Data Service can support you in creating a data dictionary, and create any <code>Schemas</code> for your research group. Contact <a href=mailto:c.seal@auckland.ac.nz>Chris Seal</a> for more information.</p> </div> <p>In the Instrument Data Wizard, first select the <em>Project</em>, <em>Experiment</em>, <em>Dataset</em> or <em>Datafile</em> you wish to edit, then you can change metadata on the right-hand pane.</p> <p><img alt=image src=../../../assets/metadata-2.png></p> <p>The <code>Description</code> tab contains the basic metadata fields, while the <code>Metadata</code> tab contains the <code>Schema</code> metadata fields.</p> <p>At the moment, the Instrument Data Wizard accepts free-text <code>Parameter</code> names and values.</p> <h3 id=recording-sarahs-metadata>Recording Sarah's metadata<a class=headerlink href=#recording-sarahs-metadata title="Permanent link">¶</a></h3> <p>As Sarah, you have two things you need to include in the metadata. You need to note down the instrument the sequencing was done on, and the sequence depth used. In genetics, sequence depth measures the <a href=https://en.wikipedia.org/wiki/Coverage_(genetics)>completeness of the sequencing process</a>.</p> <h4 id=adding-the-instrument-id>Adding the instrument ID<a class=headerlink href=#adding-the-instrument-id title="Permanent link">¶</a></h4> <p>To record the instrument, you first need to find the instrument's <a href=https://datascience.codata.org/articles/10.5334/dsj-2020-018>persistent identifier (PID)</a>. For Sarah, the sequencing company has given her the ID <code>http://hdl.handle.net/21.T11998/0000-001A-3904-0</code>.</p> <div class="admonition question"> <p class=admonition-title>Where can I find my instrument's persistent identifier?</p> <p>You can log in to the Instrument Data Service web portal to find the ID. See <a href=../../04-finding-instrument-id/ >here</a>.</p> </div> <p>Instrument is a basic metadata field, so you can find it in the <code>Description</code> tab, as the <code>Instrument ID</code> field.</p> <p>After filling out the field, the editor should look like this.</p> <p><img alt=image src=../../../assets/metadata-3.png></p> <h4 id=adding-the-sequence-depth>Adding the sequence depth<a class=headerlink href=#adding-the-sequence-depth title="Permanent link">¶</a></h4> <p>You have decided with your team that sequence depth should be recorded as a <code>Schema Parameter</code> on each <code>Experiment</code>, with the name <code>Depth of sequencing</code>, and value as an integer.</p> <p>Try adding 100 as the sequence depth for the Herceptin <code>Experiment</code>. Once finished, your editor should look like this.</p> <p><img alt=image src=../../../assets/metadata-1.png></p> <p>If you need to delete a <code>Parameter</code> row, select that row, then click the <code>Remove</code> button.</p> <div class="admonition info"> <p class=admonition-title>Keep your <code>Schema</code> names and values consistent</p> <p>Record <code>Parameter</code> names and values consistently, using the same letter casing and units. This will help with finding your data in the future. For example, if you have a <code>Parameter</code> representing a length, decide on the name (e.g. "distance", no uppercase) and the value unit (e.g. millimetre), and use them consistently.</p> </div> <h3 id=exercise-adding-more-metadata>Exercise: Adding more metadata<a class=headerlink href=#exercise-adding-more-metadata title="Permanent link">¶</a></h3> <ol> <li> <p><code>Lead Researcher</code> is another basic metadata field on <code>Projects</code>. Try adding yourself as the lead researcher in the <code>Project</code>, using Sarah's University username <code>skau921</code>.</p> <p>Your editor should look like this:</p> <p><img alt=image src=../../../assets/metadata-4.png></p> </li> <li> <p>For the Herceptin experiment, there was an error in the sequencing process. You would like to mark it as inaccurate. Decide on how you would represent this, then annotate the experiment.</p> <p>This is one way you may like to add this:</p> <p><img alt=image src=../../../assets/metadata-5.png></p> </li> </ol> </div></section><section class=labstep data-step=5><div class=instructions><h2 id=restrict-access-to-your-data class=step-title>5. Restrict access to your data<a class=headerlink href=#restrict-access-to-your-data title="Permanent link">¶</a></h2> <p>In the Instrument Data Wizard, you can define which users or groups of users have access to your data in the Instrument Data Service. </p> <p>Instrument Data Service has a cascading access control mechanism which allows you to apply restrictions on which users or groups of users can access a file or a group of files. Permissions granted at a higher level of the hierarchy will cascade down to all the files nested in it unless explicitly overridden.</p> <p>Users must have an account created in Instrument Data Service in order to access the data.</p> <div class="admonition example"> <p class=admonition-title>Adding Restrictions on your own data</p> <p>Think about who needs to be able to access your data. What kind of access should they have? If there's a subset of the data that should be accessible to more or fewer people, how should they be organised?</p> </div> </div></section><section class=labstep data-step=6><div class=instructions><h2 id=save-the-ingestion-file class=step-title>6. Save the ingestion file<a class=headerlink href=#save-the-ingestion-file title="Permanent link">¶</a></h2> <p>Once you are finished organising and annotating your data, you can save the ingestion file for the last time. Ensure to save the ingestion file in the root of your data folder.</p> <p>Your data folder should look like this.</p> <p><img alt=image src=../../../assets/export-1.png></p> <p>We are finished! When your data is copied and ingested into Instrument Data Service , the ingestion file will be examined for a list of data files, how they should be structured, metadata associated with them, and access control properties. If there are any errors or validation problems in the process, you will be notified.</p> <p>This is the end of this introductory tutorial. Please contact <a href=mailto:c.seal@auckland.ac.nz>Chris Seal</a> for any further questions.</p></div></section> </article> <style>
.stepper{
  width:calc(100% - var(--sidebar-width));
  position: fixed;
  right:0;
  bottom:2rem;
  padding:0 5vw;
  overflow: visible;
  align-items: flex-end;
}
#previous-step{
  font-size:1.2rem;
  font-weight:400;
  height:1.6em;
}
#next-step{
  font-size:1.2rem;
}
#done{
  border-radius:50%;
  width: 4em;
  height: 4em;
  position: relative;
}
#done .md-icon{
  width: 2em;
  height: 2em;
  position: absolute;
  top: 1em;
  right: 1em;
  }
#done .md-icon svg {
  height: initial;
  width: initial;
}
body[dir="rtl"] .stepper{
  right:auto;
  left:0;
}
@media screen and (max-width:76.1875em){
  .stepper{
    width:100%;
  }
}
</style> <nav class="md-footer__inner stepper" aria-label=Footer> <a id=previous-step href=# class=md-footer__link--prev aria-label=Previous: rel=prev disappear> <div class=md-icon> ↩ </div> </a> <a id=next-step href=# class=md-footer__link--next aria-label=Next: rel=next> <div class=md-icon> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"></path></svg> </div> </a> <a id=done href class=md-footer__link--next aria-label="Exit to Home" rel=done hidden> <div class=md-icon> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M21 7 9 19l-5.5-5.5 1.41-1.41L9 16.17 19.59 5.59 21 7Z"></path></svg> </div> </a> </nav> </div> </div> </main> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "../../../../../..", "features": ["navigation.tabs", "navigation.tabs.sticky", "navigation.tracking", "navigation.top", "navigation.indexes", "toc.follow", "navigation.top", "search.suggest", "search.highlight", "content.tabs.link", "content.code.annotation", "content.code.copy", "content.action.edit"], "search": "../../../../../../assets/javascripts/workers/search.f886a092.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script> <script src=../../../../../../assets/javascripts/bundle.6c14ae12.min.js></script> <script>
    window.requestAnimationFrame(
      function(){
        let u = document.querySelector('[unresolved]');
        u.removeAttribute("unresolved");
      });
  </script> <script src=../../../../../../assets/javascripts/codelab.js></script> </body> </html>